### ISSUE TITLE
B as alias is not parsed

### ISSUE THREAD

**Issue by quantumkoen** [QE-Lab/libqasm#56]
This code from the qx simulator test suite is not parsed:

```
version 1.0 

# file   : full_adder.qc
# author : Nader Khammassi
# brief  : quantum full adder (carry-save addition)

# define 4 qubits
qubits 4

# rename qubits
map q[0],A 
map q[1],B
map q[2],C
map q[3],D


# init inputs to some values
.init
  x A 
  x C 

  display_binary  #  initial input value


# perform addition
.add

  toffoli B,C,D
  cnot    B,C 
  toffoli A,C,D
  cnot    A,C 

  display_binary #  result

.reverse_add

  cnot    A,C 
  toffoli A,C,D
  cnot    B,C 
  toffoli B,C,D

  display_binary #  result

# error_model depolarizing_channel, 0.1 
```

Reports:

```
[+] loading circuit from 'tests/circuits/full_adder.qc' ...
Error while parsing file tests/circuits/full_adder.qc: 
syntax error, unexpected BITHEAD, expecting NAME
```

**Comment by kel85uk** [comment]
Closing due to no response, and the way I interpret it is that b and q should be a special keyword for cQASM the same way as special registers (e.g. ax) are reserved for classical assembly.

**Comment by quantumkoen** [comment]
Given that it is trivial to distinguish between the use of b as a signifier for adressing bits vs using it as an alias, and given that the cqasm 1.0 syntax nowhere says b cannot be used as an alias, I think it should be supported by the parser.

**Comment by kel85uk** [comment]
True, reopening this.

**Comment by jvanstraten** [comment]
This has been supported since #105, though due to a bug I stumbled on and fixed in #115 I think that particular mapping has only worked since then. Note that you wouldn't be able to refer to the measurement bits associated with the qubits once you redefine `b` though (unless you make another remap `b` first).
